autofs-5.1.9 - handle sss special case getautomntbyname() error

From: Ian Kent <raven@themaw.net>

The sss key lookup (via getautomntbyname()) returns EHOSTDOWN when the
entry is invalid, such as when the location is empty. But setatomntent()
has already been called successfully so we know the host is up and the
map exists hence this probably should be EINVAL.

In both these cases the better return is NSS_STATUS_UNAVAIL.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG            |    1 +
 modules/lookup_sss.c |    6 +++---
 2 files changed, 4 insertions(+), 3 deletions(-)

--- autofs-5.1.4.orig/CHANGELOG
+++ autofs-5.1.4/CHANGELOG
@@ -166,6 +166,7 @@
 - always recreate credential cache.
 - fix always recreate credential cache.
 - fix missing unlock in sasl_do_kinit_ext_cc().
+- handle sss special case getautomntbyname() error.
 
 xx/xx/2018 autofs-5.1.5
 - fix flag file permission.
--- autofs-5.1.4.orig/modules/lookup_sss.c
+++ autofs-5.1.4/modules/lookup_sss.c
@@ -658,8 +658,8 @@ static int getautomntbyname(unsigned int
 				err = NSS_STATUS_NOTFOUND;
 				goto free;
 			}
-			if (ret != EHOSTDOWN)
-				goto error;
+			if (ret == EINVAL || ret == EHOSTDOWN)
+				goto free;
 		}
 
 		ret = getautomntbyname_wait(logopt, ctxt,
@@ -670,7 +670,7 @@ static int getautomntbyname(unsigned int
 			if (ret == ETIMEDOUT)
 				goto error;
 			/* sss proto version 0 and sss timeout not set */
-			if (ret == EINVAL)
+			if (ret == EINVAL || ret == EHOSTDOWN)
 				goto free;
 			if (ret == ENOENT) {
 				err = NSS_STATUS_NOTFOUND;
